### PR TITLE
Pin `pytest<8` for pypy

### PR DESCRIPTION
**Pull request by jtpio**
## References

Investigating https://github.com/jupyterlite/jupyterlite/issues/1308

### CODE REVIEW

**Comment by jtpio** [comment]
Pinning on `pytest<8` seems to be doing the trick:

image

We should probably keep https://github.com/jupyterlite/jupyterlite/issues/1308 opened, or open a new issue to remember looking at that again later.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -168,6 +168,9 @@ jobs:
           jupyter lite status --debug || exit 1
       - name: Setup pip (test)
         run: ${{ matrix.python-command }} -m pip install -r requirements-test.txt
+      - name: Pin pytest for pypy
+        if: ${{ matrix.python-version == 'pypy-3.8' }}
+        run: ${{ matrix.python-command }} -m pip install "pytest<8"
```
Comment (bollwyvl) [inline .github/workflows/build.yml:173]: Could write something (e.g. link to issue) on `$GITHUB_STEP_SUMMARY`...

```
echo 'See https://github.com/jupyterlite/jupyterlite/issues/1308' >> $GITHUB_STEP_SUMMARY
```

Comment (jtpio) [reply .github/workflows/build.yml:173]: Sure let's try that.

Comment (jtpio) [reply .github/workflows/build.yml]: Done in https://github.com/jupyterlite/jupyterlite/pull/1309/commits/6a6dbe36e5a8425e6af27e0fe4826a0a1c32f2a5

**Pull request merged**